core/src/test/scala/unit/kafka/tools/ConsoleProducerTest.scala

```diff
@@ -140,6 +141,29 @@ class ConsoleProducerTest {
     assert(reader.parseKey)
   }
 
+  @Test
+  def testParseConfigFile (): Unit = {
+    val propsFile = TestUtils.tempFile()
+    val propsStream = Files.newOutputStream(propsFile.toPath)
+    propsStream.write("parse.key=true\n".getBytes())
+    propsStream.write("key.separator=|".getBytes())
+    propsStream.close()
+
+    val args = Array(
+      "--bootstrap-server", "localhost:9092",
+      "--topic", "test",
+      "--property", "key.separator=;",
+      "--property", "parse.headers=true",
+      "--reader-config", propsFile.getAbsolutePath
+    )
+    val config = new ConsoleProducer.ProducerConfig(args)
+    val reader = Class.forName(config.readerClass).getDeclaredConstructor().newInstance().asInstanceOf[LineMessageReader]
+    reader.init(System.in,ConsoleProducer.getReaderProps(config))
```

Comment:
nit: We need to add a space after `,`.

Reply:
Fixed.
Also applied on code of previous test.